homeassistant/components/itach/remote.py

```diff
@@ -106,22 +117,22 @@ def is_on(self):
     def turn_on(self, **kwargs):
         """Turn the device on."""
         self._power = True
-        num_repeats = kwargs.get(ATTR_NUM_REPEATS, 1)
-        self.itachip2ir.send(self._name, "ON", num_repeats)
+        self.itachip2ir.send(self._name, "ON", self._ir_count)
         self.schedule_update_ha_state()
 
     def turn_off(self, **kwargs):
         """Turn the device off."""
         self._power = False
-        num_repeats = kwargs.get(ATTR_NUM_REPEATS, 1)
-        self.itachip2ir.send(self._name, "OFF", num_repeats)
+        self.itachip2ir.send(self._name, "OFF", self._ir_count)
```

Comment:
You need to still respect `NUM_REPEATS` when it is passed in

```python
num_repeats = kwargs.get(ATTR_NUM_REPEATS, self._ir_count)
```

Reply:
So prior to PR 36362, NUM_REPEATS was never honored, neither for turn_on/off nor for send_command.  I don’t think one should ever intend to repeat a turn_on or turn_off command, unless the device ‘needs’ to see a command repeated in order to respond, which was the driver for implementing this ‘ir_count’ parameter. 

We can honor the NUM_REPEATS for turn_on/off, but I can’t think of a valid use case.  Note that ‘send_command’ does in fact honor NUM_REPEATS, which can be useful for smooth volume adjustment among other things.